src/http-client.ts

```diff
@@ -23,24 +23,22 @@ export class HttpClient {
     504
   ]);
 
-  async requestApiWithRetry (path: string, opts: RequestApiOptions, retryInterval?: number): Promise<any> {
-    const request = retryPromise<any>(
+  requestApiWithRetry<T = any> (path: string, opts: RequestApiOptions, retryInterval?: number): RetryPromise<T> {
```

Comment:
I changed this to return the `RetryPromise` and not just the `Promise` so callers could have more control over if/when to cancel the request. By just returning the promise, the caller would have to wait for the promise to finish (which could be indefinitely).